lld/ELF/LinkerScript.h

```diff
@@ -413,6 +422,11 @@ class LinkerScript final {
     PotentialSpillSection *tail;
   };
   llvm::DenseMap<InputSectionBase *, PotentialSpillList> potentialSpillLists;
+
+  // Named lists of input sections that can be collectively referenced in output
+  // section descriptions. Multiple references allow for sections to spill from
+  // one output section to another.
+  llvm::StringMap<SectionClassDesc *> sectionClasses;
```

Comment:
`DenseMap<CachedHashStringRef, >` or `<StringRef, >`

This header otherwise doesn't use StringMap